mlx/c/closure.cpp

```diff
@@ -183,7 +183,7 @@ extern "C" mlx_closure_value_and_grad mlx_closure_value_and_grad_new_func(
             "mlx_closure_value_and_grad returned a non-zero value");
       }
       auto cpp_res =
-          std::tie(mlx_vector_array_get_(res_0), mlx_vector_array_get_(res_1));
+          std::make_pair(mlx_vector_array_get_(res_0), mlx_vector_array_get_(res_1));
```

Comment:
Sorry this is not really documented, but this file is auto-generated, so one should not modify it manually. Instead, modify `mlxtypes.py`

```
cpp_make_tuple = "std::make_pair" if n == 2 else "std::make_tuple"
    types.append(
        {
            "cpp": cpp_tuple + "<" + ", ".join(cpp_types) + ">",
            "alt": [cpp_tuple + "<" + ", ".join(alt_types) + ">"] + alts,
            "c_to_cpp": lambda s: cpp_make_tuple + "("
```

The file can then be generated via
```
python ../../python/closure_generator.py --implementation > closure.cpp
```

Reply:
Ah OK.

Reply:
I added the change to the `mistypes.py` code generator, but if I run it there are a lot more changes than I would expect (I would have expected zero - keeping my manual change).

Reply:
Ah - got you message on `clang-format` now. Now executed.